ocaml/xapi/xapi_observer.ml

```diff
@@ -249,6 +253,146 @@ end
 let startup_components =
   List.filter (( <> ) Component.Xapi_clusterd) Component.all
 
+module ObserverConfig = struct
+  type t = {
+      otel_service_name: string
+    ; otel_resource_attributes: (string * string) list
+    ; otel_exporter_zipkin_endpoints: string list
+    ; xs_exporter_bugtool_endpoint: string option
+  }
+
+  let zipkin_endpoints endpoints =
+    (*For now, this accepts all endpoints except bugtool. This will change as more endpoints are added*)
+    List.filter
+      (fun endpoint ->
+        match Tracing.endpoint_of_string endpoint with
+        | Tracing.Url _ ->
+            true
+        | _ ->
+            false
+      )
+      endpoints
+
+  let rec bugtool_endpoint endpoints =
+    match endpoints with
+    | x :: _ when Tracing.(endpoint_of_string x = Bugtool) ->
+        Some x
+    | x :: t ->
+        bugtool_endpoint t
+    | [] ->
+        None
+
+  let attributes_to_W3CBaggage attrs =
+    List.map
+      (fun (k, v) -> (k, Tracing.W3CBaggage.Value.(v |> serialise |> to_string)))
+      attrs
+
+  let config_of_observer ~__context ~component ~observer =
+    let endpoints = Db.Observer.get_endpoints ~__context ~self:observer in
+    {
+      otel_service_name= component
+    ; otel_resource_attributes=
+        attributes_to_W3CBaggage
+          (Db.Observer.get_attributes ~__context ~self:observer)
+    ; otel_exporter_zipkin_endpoints= zipkin_endpoints endpoints
+    ; xs_exporter_bugtool_endpoint= bugtool_endpoint endpoints
+    }
+end
+
+module type OBSERVER_COMPONENT = sig
+  val component_name : string
```

Comment:
instead of using `component_name : string`, it's better to use `component: Component.t` here